modules/machineset-gcp-confidential-vm.adoc

```diff
@@ -0,0 +1,55 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating_machinesets/creating-machineset-gcp.adoc
+// * machine_management/control_plane_machine_management/cpmso-using.adoc
+
+ifeval::["{context}" == "cpmso-using"]
+:cpmso:
+endif::[]
+
+:_content-type: PROCEDURE
+[id="machineset-gcp-confidential-vm_{context}"]
+= Configuring Confidential VM by using machine sets
+
+By editing the machine set YAML file, you can configure the Confidential VM options that a machine set uses for machines that it deploys.
+
+For more information about Confidential Compute features, functionality, and compatibility, see the GCP Compute Engine documentation about link:https://cloud.google.com/compute/confidential-vm/docs/about-cvm[Confidential VM].
+
+.Procedure
+
+. In a text editor, open the YAML file for an existing machine set or create a new one.
+
+. Edit the following section under the `providerSpec` field:
++
+[source,yaml]
+----
+ifndef::cpmso[]
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+endif::cpmso[]
+ifdef::cpmso[]
+apiVersion: machine.openshift.io/v1
+kind: ControlPlaneMachineSet
+endif::cpmso[]
+...
+spec:
+  template:
+    spec:
+      providerSpec:
+        value:
+          confidentialCompute: Enabled <1>
+          onHostMaintenance: Terminate <2>
+          machineType: n2d-standard-8 <3>
+...
+----
+<1> Specify whether Confidential VM is enabled. Valid values are `Disabled` or `Enabled`.
+<2> Specify the behavior of the VM during a host maintenance event, such as a hardware or software update. For a machine that uses Confidential VM, this value must be set to `Terminate`, which stops the VM. Confidential VM does not support live VM migration.
+<3> Specify a machine type that supports Confidential VM.
```

Comment:
Worth mentioning that Confidential VM supports compute engine machine types in the following series:
N2D, C2D
https://cloud.google.com/compute/confidential-vm/docs/os-and-machine-type#machine-type

Reply:
good call :) added that and will send to QE. Thanks for your review!